core/src/main/scala/com/microsoft/azure/synapse/ml/causal/LinearDMLEstimator.scala

```diff
@@ -0,0 +1,330 @@
+import org.apache.spark.ml.regression.{GeneralizedLinearRegression, Regressor}
+
+import scala.concurrent.Future
+
+/** Linear Double ML estimators. The estimator follows the two stage process,
+ *  where a set of nuisance functions are estimated in the first stage in a crossfitting manner
+ *  and a final stage estimates the average treatment effect (ATE) model.
+ *  Our goal is to estimate the constant marginal ATE Theta(X)
+ *
+ *  In this estimator, the ATE is estimated by using the following estimating equations:
+ *  .. math ::
+ *      Y - \\E[Y | X, W] = \\Theta(X) \\cdot (T - \\E[T | X, W]) + \\epsilon
+ *
+ *  Thus if we estimate the nuisance functions :math:`q(X, W) = \\E[Y | X, W]` and
+ *  :math:`f(X, W)=\\E[T | X, W]` in the first stage, we can estimate the final stage ate for each
+ *  treatment t, by running a regression, minimizing the residual on residual square loss,
+ *  estimating Theta(X) is a final regression problem, regressing tilde{Y} on X and tilde{T})
+ *
+ *  .. math ::
+ *       \\hat{\\theta} = \\arg\\min_{\\Theta}\
+ *       \E_n\\left[ (\\tilde{Y} - \\Theta(X) \\cdot \\tilde{T})^2 \\right]
+ *
+ * Where
+ * `\\tilde{Y}=Y - \\E[Y | X, W]` and :math:`\\tilde{T}=T-\\E[T | X, W]` denotes the
+ * residual outcome and residual treatment.
+ *
+ * The nuisance function :math:`q` is a simple regression problem and user
+ * can use setOutcomeModel to set an arbitrary sparkml regressor that is internally used to solve this regression problem
+ *
+ * The problem of estimating the nuisance function :math:`f` is also a regression problem and user
+ * can use setTreatmentModel to set an arbitrary sparkml regressor that is internally used to solve this regression problem.
+ *
+ * The input categorical treatment is one-hot encoded (excluding the lexicographically smallest treatment which is used as the baseline)
+ * and the `predict_proba` method of the treatment model classifier is used to residualize the one-hot encoded treatment.
+
+      The final stage is (potentially multi-task) linear regression problem with outcomes the labels
+      :math:`\\tilde{Y}` and regressors the composite features
+      :math:`\\tilde{T}\\otimes \\phi(X) = \\mathtt{vec}(\\tilde{T}\\cdot \\phi(X)^T)`.
+      The :class:`.DML` takes as input parameter
+      ``model_final``, which is any linear sparkml regressor that is internally used to solve this
+      (multi-task) linear regression problem.
+ */
+//noinspection ScalaStyle
+class LinearDMLEstimator(override val uid: String)
+  extends Estimator[LinearDMLModel] with ComplexParamsWritable
+    with LinearDMLParams with BasicLogging with Wrappable {
+
+  logClass()
+
+  def this() = this(Identifiable.randomUID("LinearDMLEstimator"))
+
+  /** Fits the LinearDML model.
+   *
+   * @param dataset The input dataset to train.
+   * @return The trained LinearDML model, from which you can get Ate and Ci values
+   */
+  override def fit(dataset: Dataset[_]): LinearDMLModel = {
```

Comment:
Please add one more check in fit: getMaxIter must be greater than zero. The parameter defined in the `HasMaxIter` trait has validation logic as `ParamValidators.gtEq(0)`, but zero iteration will result in no ATE estimates, and it doesn't make sense.